### DISCUSSION TITLE
Feature request: A proper projectile reflector.

### DISCUSSION THREAD

**Discussion by VinVulpis** [ikemen-engine/Ikemen-GO#898] [Feature Request]
If at all possible, it would be neat to have a more proper universal reflect STCTRL (or maybe an attribute added to Reversaldef) that can recognize both Projdef and helper based projectiles. 

Maybe have an attribute you can give helpers to flag them AS a projectile too to make it simpler for what the STCTRL is looking for maybe.

Something for a character to claim ownership of the "projectile" entity without messing up the attributes of them.

Obviously making a closed game this can be done by making every projectile a common state, but for the sake of testing concepts against other content, or just mixing other creations in a build, would be neat to have this.

**Comment by potsmugen** [comment]
I'm not sure how a dedicated reflector sctrl would work. But existing features already allow proper projectile reflection. My Lei-Lei char makes use of them.
